src/components/v2/Layout/Header/index.tsx

```diff
@@ -1,27 +1,30 @@
 /** @jsxImportSource @emotion/react */
 import React, { useMemo } from 'react';
-import { RouteComponentProps, withRouter } from 'react-router-dom';
+import { useLocation } from 'react-router-dom';
 import AppBar from '@mui/material/AppBar';
 import Box from '@mui/material/Box';
+
+import { useTranslation } from 'translation';
 import { Toolbar } from '../Toolbar';
 import ConnectButton from './ConnectButton';
 import { XvsCoinInfo, VaiCoinInfo } from '../CoinInfo';
-import { useStyles } from './styles';
 import { menuItems } from '../constants';
+import { useStyles } from './styles';
 
-type HeaderProps = RouteComponentProps;
-
-const Header = ({ location }: HeaderProps) => {
-  const title = useMemo(
-    () => menuItems.find(item => item.href === location.pathname)?.text,
-    [location.pathname],
-  );
+const Header = () => {
+  const location = useLocation();
   const styles = useStyles();
+  const { t } = useTranslation();
+
+  const title = useMemo(() => {
+    const currentItemId = menuItems.find(item => item.href === location.pathname)?.id;
+    return t(`layout.menuItems.${currentItemId}`);
+  }, [location.pathname]);
```

Comment:
There's two issues with that unfortunately:
- it's risky, because there's no check done by the `t` function to verify the key passed exists in the translations
- automatically extracting the translation keys by running `yarn extract-translation-keys` will not catch these. Even worse, it will delete them from the translation file if you added them manually

The solution for this is to include the text directly inside each menu item.